lms/djangoapps/lti_provider/views.py

```diff
@@ -64,8 +66,14 @@ def lti_launch(request, course_id, usage_id):
     # Store the course, and usage ID in the session to prevent privilege
     # escalation if a staff member in one course tries to access material in
     # another.
-    params['course_id'] = course_id
-    params['usage_id'] = usage_id
+    try:
+        course_key, usage_key = parse_course_and_usage_keys(course_id, usage_id)
+    except InvalidKeyError:
+        raise Http404(
+            'Invalid course key {} or usage key {}'.format(course_id, usage_id)
```

Comment:
If it's going to be a user-facing error message, then:

1. I think this might confuse users, and we might be better off with a more generic "content could not be found" type of message, and logging the details.
2. We'd need to mark the string for translation.